source/features/extend-file-types-filter.tsx

```diff
@@ -0,0 +1,216 @@
+ * @returns {HTMLElement[]} each file detail header element
+ */
+const extendPRFileElements = (): HTMLElement[] =>
+	select.all('.file.Details').map(elem => {
+		const fileHeaderElem: HTMLElement = elem.querySelector('.file-header');
+		const fileExt = getFullExt(fileHeaderElem.dataset.path);
+		fileHeaderElem.dataset.fileType = fileExt || '';
+		return fileHeaderElem;
+	});
+
+/**
+ * Gets all PR files on the page. Uses the DOM tree to find the files.
+ * As well as the deleted status from the file DOM element
+ * @returns {PRFile[]} list of each pull request files info
+ */
+const getPRFilesFromDom = (): PRFile[] =>
+	extendPRFileElements().map(elem => {
+		return {
+			deleted: elem.dataset.fileDeleted === 'true',
+			fileExt: elem.dataset.fileType
+		};
+	});
+
+/**
+ * Gets all files from pull request via the github api.
+ * @returns {PRFile[]} list of each pull request files info
+ */
+const getPRFilesFromApi = async (): Promise<PRFile[]> => {
+	const pullUrl = getCleanPathname().replace('/pull/', '/pulls/');
+	const apiUrl = `repos/${pullUrl}?per_page=${fileListLimit}`;
+	try {
+		const result = await api.v3(apiUrl);
+		return result.map(({status, filename}) => {
+			return {
+				deleted: status === deletedStatus,
+				fileExt: getFullExt(filename)
+			};
+		});
+	} catch (error) {
+		return null;
+	}
+};
+
+/**
+ * Group PR files by extension aggregating total and deleted counts
+ * @param {PRFile[]} prFiles list of pr files with pertinent info
+ * @returns {PRFileExts} object map of file extensions to combined details
+ */
+const groupPRFilesByExt = (prFiles: PRFile[]): PRFileExts =>
+	prFiles.reduce((accumulator, {fileExt, deleted}) => {
+		if (!fileExt) {
+			return accumulator;
+		}
+
+		if (!(fileExt in accumulator)) {
+			accumulator[fileExt] = {count: 0, deleted: 0};
+		}
+
+		accumulator[fileExt].count += 1;
+		if (deleted) {
+			accumulator[fileExt].deleted += 1;
+		}
+
+		return accumulator;
+	}, {});
+
+/**
+ * Replace Github default filter list
+ * @param {PRFile[]} prFiles list of pr files with pertinent info
+ */
+const extendFileTypesFilter = (prFiles: PRFile[]) => {
+	const prFilesByExt = groupPRFilesByExt(prFiles);
+	const filterList: HTMLElement = select('.select-menu-list .p-2');
+
+	while (filterList.firstChild) {
+		filterList.removeChild(filterList.firstChild);
+	}
+
+	const fileExtensions = Object.keys(prFilesByExt);
+	fileExtensions.sort().forEach(fileExtension => {
+		const props = {
+			fileExtension,
+			totalCount: prFilesByExt[fileExtension].count,
+			deletedCount: prFilesByExt[fileExtension].deleted
+		};
+		filterList.append(filterOption(props));
+	});
+	filterList.append(deletedToggle({count: fileExtensions.length}));
+};
+
+/**
+ * Listen to subtree changes to Github PR files and updates filter list
+ */
+const init = async () => {
+	const prFiles = await getPRFilesFromApi();
+	const prFilesGetter = prFiles ? () => prFiles : getPRFilesFromDom;
```

Comment:
Only get them from the API

Reply:
Sure, API relies on the user adding an api token though

Reply:
A lot of features do; that's fine